src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        DetailAST leftCurly;
+        if (ast.findFirstToken(TokenTypes.SLIST) == null) {
+            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        }
+        else {
+            leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+        }
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
```
$ cat Test.java 
public class Test {
    void method1(int a) {
        switch (a) {case 1: ; } // no violation
        switch (a) {case 1:{}}  // 1 violation
   }
}

$ java -jar checkstyle-7.6-all.jar -t Test.java | grep "\[4:"
    |       |--LITERAL_SWITCH -> switch [4:8]
    |       |   |--LPAREN -> ( [4:15]
    |       |   |--EXPR -> EXPR [4:16]
    |       |   |   `--IDENT -> a [4:16]
    |       |   |--RPAREN -> ) [4:17]
    |       |   |--LCURLY -> { [4:19]
    |       |   |--CASE_GROUP -> CASE_GROUP [4:20]
    |       |   |   |--LITERAL_CASE -> case [4:20]
    |       |   |   |   |--EXPR -> EXPR [4:25]
    |       |   |   |   |   `--NUM_INT -> 1 [4:25]
    |       |   |   |   `--COLON -> : [4:26]
    |       |   |   `--SLIST -> SLIST [4:27]
    |       |   |       `--SLIST -> { [4:27]
    |       |   |           `--RCURLY -> } [4:28]
    |       |   `--RCURLY -> } [4:29]
  
```

Ok, so here is reason why we do not understand each other.
LITERAL_CASE - never have SLIST under it.
This is mistake in configuration of Check. LITERAL_CASE token should be removed from Check and CASE_GROUP is used. from CASE_GROUP you will find SLIST by the same logic as for all other tokens.

@ps-sp , please address my point.